### ISSUE TITLE
guacamole-auth-pam

### ISSUE THREAD

**Issue by bmullan** [voegelas/guacamole-auth-pam#3]
I thought I'd try guacamole-auth-pam with guacamole v0.9.14, tomcat8, freerdp-x11 and ubuntu 18.04
I think I have the extension built ok and it is in /etc/guacamole/extensions
I am using xrdp on the ubuntu host so I can connect using RDP from guacamole.

I can log into the host using a xfreerdp from a different machine and I see the xfce4 desktop

If I attempt to access via guacamole:

http://x.x.x.x:8080/guacamole

I see the TAB Icon on the browser show the Guacamole emblem but then I don't see anything further ?

I guess I expected to see a login prompt where I'd enter a UserID and UserPswd from the ubuntu 18.04 system I am trying to access.  Of course I am assuming that the guacamole-auth-pam extention   is what presents that yes/no ??

**Comment by voegelas** [comment]
The login dialog is presented by Guacamole. The guacamole-auth-pam extension is later used to verify the entered credentials and to show the available remote desktops. Make sure that the file "/etc/guacamole/unix-user-mapping.xml" exists, contains valid XML and is readable by Tomcat.

I've added a wiki page that shows [how to build Debian packages for Guacamole](https://github.com/voegelas/guacamole-auth-pam/wiki/Build-your-own-Debian-packages).

**Comment by bmullan** [comment]
@voegelas 
Andreas - I noticed that you called the file "_unix-user-mapping.xml_" instead of _user-mapping.xml_
as per [the Guacamole Documentation](https://guacamole.apache.org/doc/gug/configuring-guacamole.html#user-mapping)

Just wanted to make sure that "unix-user-mapping.xml" is what you meant?

Brian

**Comment by voegelas** [comment]
The _user-mapping.xml_ file is only used by Guacamole's built-in default authentication. The guacamole-auth-pam extension requires a different configuration file named _unix-user-mapping.xml_. The format is based on the no longer existing [guacamole-auth-noauth](https://github.com/apache/guacamole-client/blob/de12b683d746129ddc8b34425ed6e40b618c91d6/extensions/guacamole-auth-noauth/doc/example/noauth-config.xml) extension and can be parsed more easily than _user-mapping.xml_. For example, there can only be one protocol per configuration so it makes sense to use an XML attribute instead of an element. When I wrote guacamole-auth-pam both formats existed and I decided to follow the more sensible format.
